pkg/apis/core/v1alpha1/federatedtypeconfig_types.go

```diff
@@ -59,17 +55,35 @@ type APIResource struct {
 	// the fields enforces them as mandatory.
 
 	// Group of the resource.
-	Group string `json:"group,omitempty"`
+	Group string `json:"group"`
```

Comment:
Core kube types won't have a group.  Also, what is the point of requiring a string when an empty string satisfies the condition?

Reply:
Agree, this should be `omitempty`

Reply:
The core k8s types define the group as the empty string `""`, which is still valid for a required field, and is different from not supplying the field at all. Do we really want it to be `omitempty`?

Reply:
I don't get the point of requiring a field for which an acceptable value is the default if not supplied.

@pmorie WDDS (what does david say)?

Reply:
Found some [precedent in the RBAC API](https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/api/rbac/v1/types.go#L81)

It looks like the RBAC API has a similar field that is `omitempty` and provides a default.

Reply:
This is also used for `FederatedType` which requires the `Group` field be set. I think that supersedes the requirement to make this field required. Thoughts?